File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/MacDialog.cs
namespace AbstractFactoryDesignPattern.Example1;

public class MacDialog : Dialog
{
	public override void Render()
	{
		Console.WriteLine("Rendering Mac Dialog");
	}
	public override void Show()
	{
		Console.WriteLine("Showing Mac Dialog");
	}
	public override void Close()
	{
		Console.WriteLine("Closing Mac Dialog");
	}
}

File: java/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/main/java/creational/abstractfactory/example1/models/concretes/MacDialog.java
package creational.abstractfactory.example1.models.concretes;

import creational.abstractfactory.example1.models.abstracts.Dialog;

public class MacDialog extends Dialog {
    public void render() { System.out.println("Rendering Mac Dialog"); }
    public void show()   { System.out.println("Showing Mac Dialog"); }
    public void close()  { System.out.println("Closing Mac Dialog"); }
}
